ArjunaJTS/jtax/tests/classes/com/hp/mwtests/ts/jta/jts/jca/XATerminatorImpleUnitTest.java

```diff
@@ -242,6 +228,10 @@ public void rollback(Xid xid) throws XAException {
         xa.recover(XAResource.TMSTARTRSCAN);
         try {
             xa.commit(xid, false);
+            Assert.fail("Expecting XAException being thrown indicating more recover to be called");
+        } catch (XAException expected) {
+            Assert.assertTrue("On commit XAException error code indicating more recover call is expected but it's " + expected.errorCode,
+                XAException.XA_RETRY == expected.errorCode || XAException.XAER_RMFAIL == expected.errorCode);
```

Comment:
I put there the both codes as the either one is correct (by my undestanding). They mean that the commit fails in some way and is expected to be retried.

The point here is that the Narayana implementation works with `XAER_RMFAIL` (it's thrown) but WFTC works with `XA_RETRY`.

It would be ok to leave here `XAER_RMFAIL` but I thought is good to emphasize that both are ok.